lib/support/isInternalKey.js

```diff
@@ -1,24 +1,13 @@
-var __parent = require("./../internal/parent");
-var __key = require("./../internal/key");
-var __version = require("./../internal/version");
-var __prev = require("./../internal/prev");
-var __next = require("./../internal/next");
 var $absolutePath = require("./../internal/absolutePath");
+var prefix = require("./../internal/prefix");
 
 /**
- * If the key passed in is an internal key.  We will use a simple checking to
- * prevent infinite recursion on some machines.
+ * Determined if the key passed in is an internal key.
  *
- * @param {String} x -
+ * @param {String} x The key
  * @private
  * @returns {Boolean}
  */
 module.exports = function isInternalKey(x) {
-    return x === __parent ||
-        x === __key ||
-        x === __version ||
-        x === __prev ||
-        x === __next ||
-        x === "$size" ||
-        x === $absolutePath;
+    return (x === "$size") || (x === "$absolutePath") || (x && (x.charAt(0) === prefix));
```

Comment:
Right - I was assuming that change would come with your set of performance changes (as discussed yesterday). I was just making it work with the 'current' state of master code.

If your changes are in, I can make that change along with this request.